plugins/k8s/resoto_plugin_k8s/resources.py

```diff
@@ -2237,6 +2237,7 @@ class KubernetesIngress(KubernetesResource, BaseLoadBalancer):
         "public_ip_address": S("status", "loadBalancer", "ingress", default=[])[0]["ip"],
         # take the public ip of the first load balancer
         "ingress_spec": S("spec") >> Bend(KubernetesIngressSpec.mapping),
+        "backends": S("spec") >> S("rules", default=[]) >> ForallBend(S("host")),
```

Comment:
I'm afraid this is too simplistic. In AWS and GCP backends[] holds a list of strings with the IDs of the backend servers. So e.g. with ALBs it resolves all of the ALB Target Groups and finds the backend instances.
Translated to K8S it should hold the list of pods that traffic ends up on. To do that you not only need to evaluate the `rules[*].http.paths[*].backend.service.name` but also `spec.defaultBackend.service.name` (if present) and then check the `spec.selector` of each service. Then match `metadata.labels` of each pod against the selector and if it matches add the pod to the list of backends. That way you know if a loadbalancer is abandoned or not.

Reply:
Thanks for the insights!